tiledb/sm/query/query.cc

```diff
@@ -82,9 +82,10 @@ Query::Query(
     StorageManager* storage_manager,
     shared_ptr<Array> array,
     optional<std::string> fragment_name)
-    : array_shared_(array)
+    : array_resources_(array->array_resources())
+    , array_shared_(array)
     , array_(array_shared_.get())
-    , array_schema_(array->array_schema_latest_ptr())
+    , array_schema_(array->array_resources()->array_schema_latest_ptr())
```

Comment:
Why does the query need to know about this distinction, in order to make the call through `array_resources`? This increases the coupling of the classes.

Reply:
@ihnorton What distinction are you talking about?

Reply:
Why does the query code need to make this call through `array_resources()`? (ie, doesn't the `array` know the current state of the resources?)

Reply:
Ah I see the question now, thanks for clarifying... There is a reason, but let me fix the unit tests to confirm the current design first, then I can answer your question.